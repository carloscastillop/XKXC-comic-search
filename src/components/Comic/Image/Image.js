import React from 'react';
import {LazyLoadImage} from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/blur.css';
import style from '../../../styles/global.module.scss';
import PropTypes from 'prop-types';

const Image = ({imageSrc, imageTitle, imageAlt, imageCaption}) => {
    const caption = (imageCaption) ? <span className={style['sr-only']}>{imageCaption}</span> : null;
    return (
        <React.Fragment>
            <LazyLoadImage
                className={`${style['img-fluid']}`}
                src={imageSrc}
                alt={imageAlt}
                title={imageTitle}
                effect="blur"
            />
            {caption}
        </React.Fragment>
    );
};

export default Image;

Image.propTypes = {
    imageSrc: PropTypes.string,
    imageTitle: PropTypes.string,
    imageAlt: PropTypes.string,
    imageCaption: PropTypes.string
};
