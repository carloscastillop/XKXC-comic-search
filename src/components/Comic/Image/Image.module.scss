@import '../../../styles/global.module';

.imageContainer {
  position: relative;
  display: inline-block;
  border: solid 1px green;

  &:hover {
    cursor: pointer;

    .imageCaption {
      display: block;
    }
  }

  .imageCaption {
    text-align: center;
    display: none;
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);


    title, caption, alternate {
      display: block;
    }

    caption {
      text-align: center;
    }

    title {
      color: green
    }

    alternate {
      color: pink
    }
  }
}