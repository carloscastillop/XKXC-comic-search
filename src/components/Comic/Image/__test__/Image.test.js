import React from 'react';
import {render} from '@testing-library/react';
import Image from '../Image';
import renderer from 'react-test-renderer';

const image = {
    img: 'https://via.placeholder.com/150',
    title: 'image title',
    alt: 'image alt',
    caption: null
};

test('Get Loading container', () => {
    const {getByTestId} = render(<Image
        imageSrc={image.img}
        imageTitle={image.title}
        imageAlt={image.alt}
        imageCaption={image.caption}/>);

    const linkElement = getByTestId('image');
    expect(linkElement).toBeInTheDocument();
});

test('Image snapshop', () => {
    const tree = renderer
        .create(<Image
            imageSrc={image.img}
            imageTitle={image.title}
            imageAlt={image.alt}
            imageCaption={image.caption}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
