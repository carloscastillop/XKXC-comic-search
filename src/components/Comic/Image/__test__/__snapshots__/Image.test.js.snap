// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image snapshop 1`] = `
<div
  className="imageContainer"
  data-testid="image"
>
  <span
    className=" lazy-load-image-background blur"
    style={
      Object {
        "backgroundImage": "",
        "backgroundSize": "",
        "color": "transparent",
        "display": "inline-block",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className="img-fluid"
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </span>
</div>
`;
